core/http/server-spring/src/main/java/org/eclipse/rdf4j/http/server/repository/RepositoryController.java

```diff
@@ -363,4 +378,18 @@ private IRI createURIOrNull(Repository repository, String graphURI) {
 			return null;
 		return repository.getValueFactory().createIRI(graphURI);
 	}
+
+	private static QueryResult<?> distinct(QueryResult<?> qr) {
```

Comment:
Might be better to reuse `QueryResults.distinctResults` here. If necessary extend to accept `QueryResult<?> qr` as a param.